backend/layers/thirdparty/s3_provider_mock.py

```diff
@@ -55,3 +55,9 @@ def uri_exists(self, uri: str):
 
     def is_empty(self):
         return len(self.mock_s3_fs) == 0
+
+    def generate_presigned_url(self, path: str, expiration: int = 604800) -> str:
```

Comment:
These function definition are need for mypy to recognize this as using the S3ProviderInterface